test/before-tests.js

```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+const fs = require('fs').promises
+const path = require('path')
+const pem = require('pem')
+
+// Certificate properties
+const certProps = {
+  days: 1, // Validity in days
+  selfSigned: true
+}
+
+pem.createCertificate(certProps, async (error, keys) => {
+  if (error) throw error
+  await fs.writeFile(path.join(__dirname, 'https', 'fastify.cert'), keys.certificate)
+  await fs.writeFile(path.join(__dirname, 'https', 'fastify.key'), keys.serviceKey)
```

Comment:
Don't mix callbacks and promises. In real-world code this is going to either leak memory ( < node v15) or crash your program badly (node v15+). Let's not promote bad patterns.